pkg/event/target/mqtt.go

```diff
@@ -119,10 +119,6 @@ func (target *MQTTTarget) send(eventData event.Event) error {
 
 // Send - reads an event from store and sends it to MQTT.
 func (target *MQTTTarget) Send(eventKey string) error {
-	if !target.client.IsConnectionOpen() {
```

Comment:
Why are we removing this?

Reply:
These are all frivolous checks @Praveenrajmani MQTT library does the right thing. Also, we should attempt a send() and not rely prematurely on this function.  Connection establishment is a disconnected call so prematurely assuming that connection might not be present while the library is capable for reconnecting we should allow the library to do its thing.

Reply:
The current store implementation will look for this particular `errNotConnected` to decide whether to retry or log.
Ignoring this will log the connection failures.. There will be too much connection log messages if there are more events.

Checking the connection and replying back with appropriate error will make the queueStore to decide what to act upon

Reply:
This is my problem with the current implementation, it is too heavy and detail-oriented and much of the detail is never documented or lost. Anyone writing new targets will have to remember a whole bunch of verbatim to get a working notification target. 

The side effect of this is already visible consistent bugs, a developer cannot fix a bug without worrying about creating problems across the implementation. Such brittle implementations are guaranteed to fail in the long run. 

I think there are some fundamental issues with the current implementation which we should address subsequently, I will bring back the code for now.

Reply:
Since the targets are hybrid, The store implementation has some rules like these to work with. For eg, `errNotConnected` is one common error shared by all the targets to denote the connection failure. Store will look this conn err and act upon. Store is plugged in to work generically for all the targets. So, these targets and future targets should obey these rules. That is where it is getting complicated. We can reduce this subsequently.

Reply:
Design documenting this would help. Will prepare one.